src/vs/workbench/contrib/terminal/electron-browser/terminalService.ts

```diff
@@ -117,7 +117,30 @@ export class TerminalService extends BrowserTerminalService implements ITerminal
 		});
 	}
 
-	private _detectWindowsShells(): Promise<IQuickPickItem[]> {
+	/**
+	 * Get the executable file path of shell from registry.
+	 * @param Registry The data of imported from `vscode-windows-registry`
+	 * @param shellName The shell name to get the executable file path
+	 * @returns The executable file path of shell or `'ShellNotFound'`
+	 */
+	private _getShellPathFromRegistry(Registry: typeof import('vscode-windows-registry'), shellName: string): string {
```

Comment:
I think this function should be a little more targetted to just pwsh for now and then return either `[]` or `[ 'path' ]`